coderd/database/dbmem/dbmem.go

```diff
@@ -13562,6 +13578,43 @@ func (q *FakeQuerier) GetAuthorizedWorkspaces(ctx context.Context, arg database.
 			}
 		}
 
+		if arg.HasAITask.Valid {
+			hasAITask, err := func() (bool, error) {
+				build, err := q.getLatestWorkspaceBuildByWorkspaceIDNoLock(ctx, workspace.ID)
+				if err != nil {
+					return false, xerrors.Errorf("get latest build: %w", err)
+				}
+				if build.HasAITask.Valid {
+					return build.HasAITask.Bool, nil
+				}
+				// If the build has a nil AI task, check if the job is in progress
+				// and if it has a non-empty AI Prompt parameter
+				job, err := q.getProvisionerJobByIDNoLock(ctx, build.JobID)
+				if err != nil {
+					return false, xerrors.Errorf("get provisioner job: %w", err)
+				}
+				if job.CompletedAt.Valid {
+					return false, nil
+				}
+				parameters, err := q.getWorkspaceBuildParametersNoLock(build.ID)
+				if err != nil {
+					return false, xerrors.Errorf("get workspace build parameters: %w", err)
+				}
+				for _, param := range parameters {
+					if param.Name == "AI Prompt" && param.Value != "" {
```

Comment:
TODO: this should use the const I'm introducing in the provider: https://github.com/coder/terraform-provider-coder/blob/dk/coder-ai-task-res/provider/ai_task.go#L22